src/nupic/research/spatial_pooler.py

```diff
@@ -1424,16 +1422,10 @@ def _inhibitColumnsGlobal(self, overlaps, density):
     #calculate num active per inhibition area
     numActive = int(density * self._numColumns)
 
-    # argpartition returns an array where all values to the left of index
-    # numActive are less than or equal to all values to its right. Negating
-    # overlaps produces a descending-order partition.
-    winnerIndices = numpy.argpartition(-overlaps, numActive)[:numActive]
-
-    # Compatibility with nupic.core requires that the winners are sorted;
-    # however, sorting only the winners is far less expensive than sorting all
-    # columns.
-    winnerValues = overlaps[winnerIndices]
-    sortedWinnerIndices = winnerIndices[numpy.argsort(-winnerValues)]
+    # Calculate winners and stable sort for compatibility with C++
+    indices = numpy.array(xrange(self._numColumns), dtype=uintType)
+    winnerIndices = indices[numpy.argsort(overlaps, kind='mergesort')]
```

Comment:
Is `kind='mergesort'` responsible for the stable sorting? If so, put a comment before this line stating that that is why it is specified.